.github/workflows/analyze-istio.yaml

```diff
@@ -0,0 +1,30 @@
+name: analyze-istio
+
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+
+jobs:
+  check-istio:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Get Istioctl
+        run: |
+          ISTIO_VERSION=$(yq eval '. | select(.kind == "Deployment") | .spec.template.spec.containers[0].image | split(":") | .[1]' ./istio/operator/manifests.yaml)
+          # downloadIstio will now retrieve the currently installed binary, instead of latest
+          echo "Downloading Istio (v${ISTIO_VERSION})"
+          curl -sL https://istio.io/downloadIstio | ISTIO_VERSION=$ISTIO_VERSION sh -
+          # this command will exit(1) if an error is found in any yaml
+          ls -lah
+          pwd
+          # istio/operator/manifests.yaml contains an empty yaml doc, which
+          # throws off analyze, so pass it in separately
+          MANIFESTS=$(yq eval '. | select(. | has("kind"))' ./istio/operator/manifests.yaml)
+
+          ./istio-${ISTIO_VERSION}/bin/istioctl analyze -A --use-kube=false \
+            --failure-threshold ERROR $(find . -not -path "*/istio-$ISTIO_VERSION/*" \
+            -not -path "*/.git*/*" -not -path "*/clusters/*" -name "*.yaml" \
```

Comment:
Instead of this, we should parse all .yaml files with yq, extract those that have `istio` in the apiVersion and pass only these to analyze.

Reply:
I considered this, but `analyze` is capable of reporting the file and line number which generated an error, and this would be lost by piping all files through yq.  I think having a specific file and line number to refer to is pretty useful, so I tried not to break that.

I really don't like the way I had to pipe `manifests.yaml` through yq.  I consider it a bug in istio, and hope to have it patched next week, which means the fix will go out in istio 1.13 in three months.

Reply:
I'm not suggesting piping the files via yq, just use yq create a list of files that are Kubernetes manifests that Istio understands. I guess Istio fails to validate anything else but Istio CRs right?

Reply:
yeah, if the object lacks a `kind` field, istio chokes.  I didn't know yq could output file names.  Can you point to a reference or example?  What should it do in the case of multi-yaml document files, where some documents are compliant and some are not?

Reply:
> What should it do in the case of multi-yaml document files, where some documents are compliant and some are not?

I think it's ok to error out for those. Usually people bundle Kubernetes objects but they don't mix them with something else.

What's the reason for excluding `clusters`? Does Istio chokes on Flux custom resources?

Reply:
I've got `yq eval '. | select(. | has("kind")) | filename'`, but that outputs filenames once for each document in the file, and it still outputs manifests.yaml, as some of the documents in that file have a `kind` field...

Reply:
Ok I think we can leave it as it is for now, it would be great if Istio would behave more like kubeval, where you can tell it to ignore unknown kinds instead of choking.

Reply:
regarding cluster files, `gotk-sync.yaml` is causing `analyze` to choke, but I'm still trying to figure out why.

regarding erroring out for multi-document yaml files, unfortunately istio/operator/manifests.yaml is once such file.  Line 50 includes two consecutive document terminators, resulting in an empty yaml document.  'analyze' shouldn't choke on this, but it currently does, and while the fix is easy, we aren't due for another release particularly soon...

Reply:
Found it.  The leading comment followed by a document break `---` in `gotk-sync.yaml` is interpreted by our yaml parser as a blank document, having no kind information.  I'm working on a PR now to ignore empty documents, but it's dependent on a major refactor landing in 1.13, so it won't be available for downstream use until Feb.